### ISSUE TITLE
[FEATURE] Make transport method for search request configurable

### ISSUE THREAD

**Issue by thopra** [TYPO3-Solr/ext-solr#1675]
Currently, the search request is sent as GET (see https://github.com/TYPO3-Solr/ext-solr/blob/master/Classes/Search.php#L148-L153 ). 

We are appending a lot of page uids to the query which can in some cases exceed the jetty configration for the request header size:

```
2017-10-09 13:22:37.684 WARN  (qtp12209492-15) [   ] o.e.j.h.HttpParser URI is too large >8192
```

Using POST would most likely solve this issue, without the need of configuring a exceptional high value on the server side. Simply adding the method parameter when invoking \ApacheSolrForTypo3\Solr\SolrService::search in the above example seems to work just fine:

```
$response = $this->solr->search(
                $query->getQueryString(),
                $offset,
                $limit,
                $query->getQueryParameters(),
                SolrService::METHOD_POST
            );
```

So it would be great if there was a TS or EM configuration to use POST instead of GET.

**Comment by dogawaf** [comment]
Hello

Since Solarium is used, we can take advantage of the PostBigRequest plugin : https://solarium.readthedocs.io/en/stable/plugins/#postbigrequest-plugin

The plugin automaticaly transpose too big GET  request (configurable) into POST request.
The plugin could be added by default or with a signal into \ApacheSolrForTypo3\Solr\System\Solr\SolrConnection::getClient() like that: 
```
    protected function getClient($endpointKey): Client
    {
        if($this->clients[$endpointKey]) {
            return $this->clients[$endpointKey];
        }

        $client = new Client(['adapter' => 'Solarium\Core\Client\Adapter\Guzzle']);
        $client->clearEndpoints();

        // register PostBigRequest Solarium plugin
        $client->getPlugin('postbigrequest');
        
        $newEndpointOptions = $this->getNode($endpointKey)->getSolariumClientOptions();
        $newEndpointOptions['key'] = $endpointKey;
        $client->createEndpoint($newEndpointOptions, true);

        $this->clients[$endpointKey] = $client;
        return $client;
    }
```

**Comment by dogawaf** [comment]
Fixed with #2476 but the fix is not functional, see https://github.com/TYPO3-Solr/ext-solr/issues/3115
A workaround exists: https://github.com/TYPO3-Solr/ext-solr/issues/3115#issue-1074218661

**Comment by dkd-kaehm** [comment]
We'll not enable it by default, who it wants can enable it in own Services.yaml:

```yaml
  Solarium\Plugin\PostBigRequest:
    tags:
      - name: event.listener
        method: 'preExecuteRequest'
        identifier: 'myext.solr.postbigrequestfix'
        event: Solarium\Core\Event\PreExecuteRequest
```
